File: api/RestaurantApi/Models/Usuario.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace RestaurantApi.Models
{
    public enum RolUsuario
    {
        Mesero,
        Cocina,
        Caja
    }

    public class Usuario
    {
        public int Id { get; set; }
        public string Nombre { get; set; } = null!;

        [Column("Usuario")]
        public string NombreUsuario { get; set; } = null!;

        public string ClaveHash { get; set; } = null!;
        public RolUsuario Rol { get; set; }
    }
}

File: app/restaurant_app/lib/models/usuario.dart
enum RolUsuario { administrador, mesero, cocina, caja }

class Usuario {
  final int id;
  final String nombre;
  final String nombreUsuario;
  final RolUsuario rol;

  Usuario({
    required this.id,
    required this.nombre,
    required this.nombreUsuario,
    required this.rol,
  });

  factory Usuario.fromJson(Map<String, dynamic> json) {
    return Usuario(
      id: json['id'],
      nombre: json['nombre'],
      nombreUsuario: json['nombreUsuario'],
      rol: _rolFromString(json['rol']),
    );
  }

  static RolUsuario _rolFromString(String rol) {
    switch (rol.toLowerCase()) {
      case 'administrador': return RolUsuario.administrador;
      case 'mesero': return RolUsuario.mesero;
      case 'cocina': return RolUsuario.cocina;
      case 'caja': return RolUsuario.caja;
      default: return RolUsuario.mesero;
    }
  }
}